devolving_music/lib/song_scores.py

```diff
@@ -1,176 +1,191 @@
-            remove: int):
+    def check_quality(self,
+                      song_submissions_sorted: Iterable["ScoreSuite"],
+                      remove: int) -> Iterable["ScoreSuite"]:
 
         song_submissions_pruned = song_submissions_sorted
 
         # do (song_submissions_sorted[i].quality_score - mvg_avg[i])
-
         # for every song in score submission
-
         # return n indices with smallest values
-
         # these indices are to be removed
-        return song_submissions_pruned
 
-    @staticmethod
-    def get_final_list(comparison_submissions: Iterable["SongComparison"],
+        return song_submissions_pruned
 
+    def get_dict_from_keys(self, new_keys) -> "dict[int, ScoreSuite]":
+        return {key: self.song_score_dict[key] for key in new_keys}
 
-                       song_submissions: Iterable["SongSubmission"]):
+    def get_final_list(self) -> Iterable["ScoreSuite"]:
 
-        scored_submissions = SongSubmission.get_scores(
-            comparison_submissions, song_submissions)
+        scored_submissions = list(self.get_scores().values())
 
         # remove all song_submissions with no information
+        info_list = SongScores.get_info_sort(scored_submissions)
 
-        # sort by peakyness
-
-        # peaky_sorted_submissions=get_peak_sort(scored_submissions)
+        informed_list = list(filter(lambda sub: sub.info_score > 0, info_list))
 
-        # Break peaky_sorted into two bing  pre peak and post peak
+        # sort by postpeakyness
+        peaky_list = SongScores.get_peak_sort(informed_list)
+        peak_loc = int(PEAK_PROPORTION * len(peaky_list))
+        # Break peaky_sorted into two bins pre peak and post peak
+        pre_peak = peaky_list[:peak_loc]
+        post_peak = peaky_list[peak_loc:]
+        # energy_sorted
+        comeup = SongScores.get_energy_sort(pre_peak)
+        cooldown = SongScores.get_energy_sort(post_peak)
+        cooldown = cooldown[::-1]
 
-        # energy_sorted=[get_energy_sort(prepeak),reverse(get_energy_sort(postpeak))]
+        # final_list is the song submission keys properly sorted
+        final_list = comeup + cooldown
 
-        # final_list = energy_sorted
-
-        final_list = []
         return final_list
 
-    def get_quality_list(
-            final_list: Iterable["SongSubmission"],
-            length_limit=300):
+    def get_quality_list(self,
+                         length_limit=300) -> Iterable["ScoreSuite"]:
 
         final_quality_list = []
 
         # if len(final_list) is above length limit
-
         # remove_index=check_quality(final_list,len(energy_sorted)-lengthlimit)
-
         # final_quality_list=remove(energy_sorted,remove_index)
 
         return final_quality_list
+
+    ###
+    # static methods can be moved into helper function
+    ###
+    @staticmethod
+    # sorts in ascending order
+    def get_info_sort(
+            score_suite_list: Iterable["ScoreSuite"]) -> Iterable["ScoreSuite"]:
```

Comment:
```suggestion
            score_suite_list: list["ScoreSuite"]) -> list["ScoreSuite"]:
```

For `sort`/`sorted` to work, `score_suite_list` needs to actually be a `list`, not just any `Iterable`.

You're definitely returning a list, so you might as well make the strongest possible assertion about the return type.

Reply:
eh I believe this is a bit of a nit pick but okay